### PR TITLE
Feature/satellite update

### PR DESCRIPTION
**Pull request by beels**
#### What does this PR do?

Update satellite to Teletype v4.0.0

#### Provide links to any related discussion on [lines](https://llllllll.co/).

No previous discussion other than a query about the status of alternate firmwares: https://llllllll.co/t/a-users-guide-to-monome-eurorack-firmwares/18031/10  Over on lines I'm 3-foot-1.

#### How should this be manually tested?

Please let me know!  I did the update very mechanically (by forcing a set of merge conflicts), and I'm running a script right now on ansible, controlled by a new-generation grid (confirms a v4 feature) and clocked with an external metro (confirms satellite variant).  But I've never touched Teletype or satellite before, so I'm not really what to do next.  I see that there is a test suite, but for the main `module` firmware, not for ansible.

#### Any background context you want to provide?

At the moment I just want to make you aware that this branch exists and seems to work.  If you think it's worth moving forward from here, I'd appreciate some guidance on testing, and then I'll do the cleanup and doc steps as needed.

#### I have,
* [ ] updated `CHANGELOG.md` and `whats_new.md`
* [ ] updated the documentation
* [ ] updated `help_mode.c` (if applicable)
* [ ] run `make format` on each commit
* [x] run tests

### CODE REVIEW

File: ansible/live_mode.c [hunk 1]
```diff
@@ -18,8 +18,25 @@ void set_slew_icon(bool display) {}
...
+
+uint8_t get_live_sub_mode() {
+    return 0;
+}
+
 void set_live_submode(u8 submode) {}
+
+void select_dash_screen(uint8_t screen) {}
+
+void print_dashboard_value(uint8_t index, int16_t value) {}
```
Comment (scanner-darkly) [inline ansible/live_mode.c]: while the dashboard itself is not used in live mode, it's possible somebody will use `PRINT` ops simply to store values, so this needs to store values and `get_dashboard_value` needs to return them

**Comment by scanner-darkly** [comment]
great job, thanks for doing this! left one comment, other than that looks good.

i would say in terms of testing i would just post a beta version in the satellite thread and ask people to help with testing, but there is one outstanding bug i need to fix (trigger inputs are not processed correctly), i'll try to do it once we can merge your PR, this way people can test both.

also, i'd still run the existing test suite - it doesn't actually test anything under /module, only /src files, so it'll provide an extra confirmation the merge is okay.

**Comment by beels** [comment]
> ... but there is one outstanding bug i need to fix (trigger inputs are not processed correctly), ...

I'm curious, what does the incorrect processing look like?

In my test patch I observed that the trigger inputs behave differently than the inputs on Teletype.  My test patch sends a clicky line level audio feedback signal to Teletype/satellite as a clock signal.  (OK, not a well-behaved choice for a test patch, but it is the patch that made me want to finally get satellite working!)  On Teletype the clicks trigger in a nice even rhythm.  On satellite it seems like the clicks arrive in dense clumps, as if the satellite inputs had a lower trigger level and no hysteresis.  I chalked it up to some difference in hardware and slapped a trigger conditioning module in front of satellite, and everything was fine, but maybe this is not an expected difference between the Teletype and Ansible input jacks?

**Comment by scanner-darkly** [comment]
it's the bug reported here: https://llllllll.co/t/satellite-run-teletype-scenes-on-ansible/35103/24
i haven't tried reproducing it yet. there is a small chance it's hardware related but likely it's just a bug in my code.

**Comment by beels** [comment]
Restored the dashboard PRINT values.  Tested by converting one of my scenes to store its temporary variables in the dashboard.

While I was at it, I noticed that other components refer to the live mode `sub_mode` setting, so I restored the getter and setter, and restored the few places where the sub mode would be set as a side-effect of a call to a live mode function.

Changed display update reporting to reflect that the display never needs to be updated.

Re-arranged the functions in `live_mode.c` to match the order they appear in the main teletype module code.

**Review by scanner-darkly — APPROVED** [review]

**Pull request merged**